etrago/cluster/gasclustering.py

```diff
@@ -0,0 +1,418 @@
+        "length": "mean",
+    }
+    strategies.update(
+        {col: "first" for col in new_links.columns if col not in strategies}
+    )
+
+    gas_carriers = [  # This list should be replace by an automatic selection
+        "CH4",
+        "CH4_to_H2",
+        "H2_feedin",
+        "H2_ind_load",
+        "H2_to_CH4",
+        "H2_to_power",
+        "power_to_H2",
+        "central_gas_CHP",
+        "central_gas_CHP_heat",
+        "industrial_gas_CHP",
+        "rural_gas_boiler",
+        "central_gas_boiler",
+        "OCGT",
+    ]
+
+    gas_links = new_links[new_links["carrier"].isin(gas_carriers)].copy()
+
+    combinations = gas_links.groupby(["bus0", "bus1", "carrier"]).agg(strategies)
+    combinations.reset_index(drop=True, inplace=True)
+
+    combinations["buscombination"] = combinations[["bus0", "bus1"]].apply(
+        lambda x: tuple(sorted([str(x.bus0), str(x.bus1)])), axis=1
+    )
+
+    strategies.update(
+        {col: "first" for col in combinations.columns if col not in strategies}
+    )
+
+    combinations_final = combinations.groupby(["buscombination", "carrier"]).agg(
+        strategies
+    )
+
+    combinations_final.set_index("link_id", inplace=True)
+    combinations_final = combinations_final.drop(columns="buscombination")
+    io.import_components_from_dataframe(network_gasgrid_c, combinations_final, "Link")
+
+    non_gas_links = (
+        new_links[~new_links["carrier"].isin(gas_carriers)]
+        .copy()
+        .drop(columns="link_id")
+    )
+    io.import_components_from_dataframe(network_gasgrid_c, non_gas_links, "Link")
+
+    if with_time:
+        for attr, df in iteritems(network.links_t):
+            if not df.empty:
+                io.import_series_from_dataframe(network_gasgrid_c, df, "Link", attr)
+
+    return network_gasgrid_c
+
+
+def kmean_clustering_gas_grid(etrago):
+    """Main function of the k-mean clustering approach. Maps the original gas
+    network to a new one with adjustable number of nodes and new coordinates.
+    Parameters
+    ----------
+    network : :class:`pypsa.Network
+        Container for all network components.
+    n_clusters_gas : int
+        Desired number of gas clusters.
+    bus_weight_tocsv : str
+        Creates a bus weighting based on conventional generation and load
+        and save it to a csv file.
+    bus_weight_fromcsv : str
+        Loads a bus weighting from a csv file to apply it to the clustering
+        algorithm.
+    Returns
+    -------
+    network : pypsa.Network object
+        Container for the gas network components.
+    """
+
+    gas_busmap = create_gas_busmap(etrago)
+
+    network_gasgrid_c = get_clustering_from_busmap(
+        etrago.network,
+        gas_busmap,
+        bus_strategies={
+            "country": "first",
+        },
+        one_port_strategies={
+            "Generator": {
+                "marginal_cost": np.mean,
+                "capital_cost": np.mean,
+                "p_nom_max": np.max,

```

Comment:
I think you should use` np.sum` for `p_nom_max` and `p_nom_min` because you want to sum up the capacity limits.

Reply:
Thanks for the correction, I've fixed this in the the commit https://github.com/openego/eTraGo/pull/359/commits/8b7ea56190e2d833fa9e28783c3c8cd29a0a5057.